pkg/controller/v1alpha2/experiment/experiment_controller.go

```diff
@@ -58,7 +62,30 @@ func Add(mgr manager.Manager) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileExperiment{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
+	r := &ReconcileExperiment{
+		Client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+	}
+	imp := viper.GetString(consts.ConfigExperimentSuggestionName)
+	r.Suggestion = newSuggestion(imp)
+	return r
+}
+
+// newSuggestion returns the new Suggestion for the given config.
+func newSuggestion(implementation string) suggestion.Suggestion {
+	// If the flag is set in CLI, use the fake implementation.
```

Comment:
The comment is outdated.

Reply:
Thanks for your review, addressed, PTAL again https://github.com/kubeflow/katib/pull/502/commits/04b2032fc379203cca0a5f93a33ef7eda154d35d